### PR TITLE
feat: updated design-tokens to v4

### LINKED ISSUE #9807: [MOB] Update design-tokens package to version 4.0.0

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR seeks to update the `@metamask/design-tokens` repo to v4. All changes noted in the [migration doc](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md) has been factored in this PR. 

## **Related issues**

Fixes: #9807 

## **Manual testing steps**
1. Due to the majority of this PR involves snapshot changes from design token changes, refer to the Screenshots/Recordings section below for main changes

### **After**
1. Switch-related code changes due to `brandColors.white000` being replaced with `brandColors.white`
- Video - https://github.com/MetaMask/metamask-mobile/assets/14355083/bf23243d-a739-45ba-ac61-b6cd7748cf24
- [BasicFunctionality.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845)
- [LoginOptionsSwitch.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-1567bc4b322bc07b0dba96165200bee3b7eb5c78532774e0303bdb75c20ecc5d)
- [SecurityOptionToggle.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-153bae2f1c750ea4210d9969c77c5150fc9be25d8e38974b50b713c12051f2aa)
- [NetworkSelector.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-ac0e8e356fae4e0141f28e04dec7ef488eadd8c64d66bff7e7d87c44c10bc93c)
- [AdvancedSettings/index.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-a15dbc3a034e3eefb70b6caadc581dd3c18c5657be514cc50cf562ea0fae9599)
- [GeneralSettings/index.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-b3065195e3dcb05950542044836756d4ea3d4dd44c2ede8dea8f781f2997e123)
- [NotificationOptionToggle/index.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-8425c66c3c353363eed46491210aeeb995ad9d72eed1d64edd2c6bcf46014b7c)
- [NotificationsSettings/index.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-4b7280553d1b5714e43fe118576a1f96498df1759afa0a9f806ec67795c7ab04)
- [BlockaidSettings.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-6c7a220f44b7e6f3bb64d7685d22376791be23f8099e18788f9be6ce48f96a9e)
- [SecuritySettings.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-2b913eb31628daca1d527bfece703633980e523a67b95426b1a331959e899a05)

2. QR code-related code changes due to `brandColors.white000` being replaced with `brandColors.white`, or `brandColors.black000` being replaced with `brandColors.black`
- Video - https://github.com/MetaMask/metamask-mobile/assets/14355083/bbbf28a4-c2c6-4d96-a370-50c53ef95bf9
- [PaymentRequestSuccess/index.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-527f168342893a6c06bea37779e3068cfa613d30c2f9b7c685324a09a62870c6)
- [AnimatedQRCode.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-0a4b0a66565dd5b3816acb7d9a4b0dca6c32a3c6d3208c9335e7b43e6d51e3be)
- [AnimatedQRScanner.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-14ae088fe434996841100e856e29af504f4e508add2dd57fc1935fca9e023d5e)
- [ReceiveRequest/index.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-897d8ac49387ea53453a406cdceb7df3750912fe13f0cbade195ab1669a657fb)
- [AddressQRCode/index.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-5c1d83c64c5696bd31d34690e88c8ffe77d7924b2f24d743e8e14a2ba2a11f74)
- [Instruction/styles.ts](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-7b2a899563edca80b76f9a573521641086786c33debb1be19805ce3fd75f1e27)
- [QRScanner/styles.ts](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-0f8f208c7b76ec71d9d38788f02ff465f0b1a5de538c4e755d8ee032797a89b4)
- [RevealPrivateCredential/styles.ts](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-5d7c30361897cca23135a1e38a90805c4cfe244532ce9c52bcf5d8ee7b2f3182)

3. `ButtonIcon` component replacement due to disabled colors being deprecated
- Video - https://github.com/MetaMask/metamask-mobile/assets/14355083/573251a0-0142-44f0-8b1c-dc117bdeff09
- [ActivationKeys.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-e3d411092b11a87757c9381193116bfa5db053d5b335ca1bd1b8950413227863)

4. `LedgerConnect` changes due to `brandColors.white000` being replaced with `brandColors.white`
- Unable to reproduce screenshots-
- [LedgerConnect/index.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-b03c2ee9e96878075fb89e73caea12ed356a63d7a891322c398350a4e8be2617)

5. `AndroidMediaPlayer` changes due to `brandColors.white000` being replaced with `brandColors.white`
- Unable to reproduce screenshots-
- [AndroidMediaPlayer.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-19b59ab0369bdc1303aa26dd3d4aff98f20741510d072ebbb789535e842168ed)

6. `FiatOnTestnetsFriction` changes due to `warning.alternative` being deprecated. Updated icon to `warning.default`
- Video - https://github.com/MetaMask/metamask-mobile/assets/14355083/82474ceb-43a9-4421-9220-54fe27aa117e
- [FiatOnTestnetsFriction.tsx](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-614111dffd0a4ac99eb603a17db6be49405b601f405ea6752543f0dc2ef88666)

7. `SliderButton` changes due to `primary.shadow` being changed to `shadow.primary`
- Video - https://github.com/MetaMask/metamask-mobile/assets/14355083/88704327-a1e4-471f-bf51-20a7d7b2fe99
- [SliderButton/index.js](https://github.com/MetaMask/metamask-mobile/pull/9845/files#diff-3edd5d1c0df5c91c03673195d2638feff3dc65c5ae99177aa6394fc5718a19d1)

8. Overall walkthrough after upgrade
- Video - https://github.com/MetaMask/metamask-mobile/assets/14355083/c75be9b7-2a0f-4648-9b48-a2f7858549db

9. Deprecated tokens search
Screenshot 2024-06-12 at 12 06 36 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/base-components/TagBase/__snapshots__/TagBase.test.tsx.snap [hunk 1]
```diff
@@ -9,11 +9,11 @@ exports[`TagBase should render TagBase 1`] = `
     {
       "alignItems": "center",
       "alignSelf": "flex-start",
-      "backgroundColor": "#FFFFFF",
-      "borderColor": "#BBC0C5",
+      "backgroundColor": "#ffffff",
```
Comment (brianacnguyen) [inline app/component-library/base-components/TagBase/__snapshots__/TagBase.test.tsx.snap:12]: brandColor.white changes

File: app/component-library/components-temp/CustomSpendCap/CustomInput/__snapshots__/CustomInput.test.tsx.snap [hunk 2]
```diff
@@ -5,7 +5,7 @@ exports[`CustomInput should render correctly 1`] = `
   style={
     [
       {
-        "backgroundColor": "#FFFFFF",
+        "backgroundColor": "#ffffff",
```
Comment (brianacnguyen) [inline app/component-library/components-temp/CustomSpendCap/CustomInput/__snapshots__/CustomInput.test.tsx.snap:8]: brandColor.white changes

**Comment by brianacnguyen** [comment]
@SocketSecurity ignore npm/@metamask/design-tokens@2.0.0

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀 🪩 
- Checked for no remaining deprecated tokens by running [this script](https://gist.github.com/georgewrmarshall/1ca0d7044bfa3c91343cadbf2bd5c828) ✅ 

### Before

Screenshot 2024-06-12 at 2 56 40 PM

### After

Screenshot 2024-06-12 at 2 57 26 PM

**Review by Cal-L — APPROVED** [review]
LGTM. For those reviewing, just filter out the snapshot files

**Pull request merged**